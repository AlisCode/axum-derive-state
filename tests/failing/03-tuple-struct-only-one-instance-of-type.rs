//! Only one instance of a type is allowed in the state

use derive_state::State;

#[derive(State)]
pub struct MyState(String, String);

fn main() {}
