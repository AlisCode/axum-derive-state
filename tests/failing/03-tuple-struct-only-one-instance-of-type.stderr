error: Type String is used multiple times in the state

         = note: The state may only contain exactly one instance of a given type
         = help: Consider wrapping this type in a newtype, e.g. pub struct MySubstate(String);

 --> tests/failing/03-tuple-struct-only-one-instance-of-type.rs:6:28
  |
6 | pub struct MyState(String, String);
  |                            ^^^^^^
